packages/mco/components/disaster-recovery/applications/failover-and-relocate-modal/dr-policy-selector.tsx

```diff
@@ -0,0 +1,143 @@
+import { getErrorMessage } from '../../../../utils';
+import {
+  ACMToDRState,
+  FailoverAndRelocateState,
+  FailoverAndRelocateAction,
+  FailoverAndRelocateType,
+  DRPolicyType,
+} from './reducer';
+
+const getDRPolicyList = async (acmToDRState: ACMToDRState[]) => [
+  ...new Set(
+    acmToDRState.map(
+      (drpcState) => drpcState?.drPolicyControl?.spec?.drPolicyRef?.name
+    )
+  ),
+];
+
+export const DRPolicySelector: React.FC<DRPolicySelectorProps> = (props) => {
+  const { state, dispatch, t } = props;
+
+  // To avoid re-rendering when a new acmToDRState object generated with same props
+  const memoizedAcmToDRState = useDeepCompareMemoize(state.acmToDRState, true);
+
+  const [drPolicies, setDRPolicies] = React.useState<DRPolicyKind[]>([]);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const setErrorMessage = React.useCallback(
+    (errorMessage: string) => {
+      dispatch({
+        type: FailoverAndRelocateType.SET_ERROR_MESSAGE,
+        payload: {
+          drPolicyErrorMessage: errorMessage,
+        },
+      });
+    },
+    [dispatch]
+  );
+
+  const setSelected = React.useCallback(
+    (selection: DRPolicyType) => {
+      dispatch({
+        type: FailoverAndRelocateType.SET_SELECTED_DR_POLICY,
+        payload: selection,
+      });
+    },
+    [dispatch]
+  );
+
+  React.useEffect(() => {
+    // Fetch DRPolicies using DRPlacementControl
+    const fetchDRPolicies = async (acmToDRState: ACMToDRState[]) => {
+      if (!!acmToDRState.length) {
+        const drPolicyList = await getDRPolicyList(acmToDRState);
+        Promise.all(
+          drPolicyList.map(async (drPolicyName) => {
+            const drPolicy: DRPolicyKind = await k8sGet({
+              model: DRPolicyModel,
+              name: drPolicyName,
+            });
+            return drPolicy;
+          })
+        )
+          .then((drPolicyObjList) => {
+            setDRPolicies(drPolicyObjList);
+            setErrorMessage('');
+          })
+          .catch((error) => {
+            setErrorMessage(getErrorMessage(error));
+          });
+      } else {
+        setDRPolicies([]);
+        setSelected({});
+      }
+    };
+    fetchDRPolicies(memoizedAcmToDRState);
+  }, [
+    memoizedAcmToDRState,
+    dispatch,
+    setDRPolicies,
+    setSelected,
+    setErrorMessage,
+  ]);
+
+  const dropdownItems = drPolicies.map((drPolicy, index) => (
+    <DropdownItem
+      id={index.toString()}
+      key={`selected-policy-${getName(drPolicy)}`}
+      value={getName(drPolicy)}
+      component="button"
+    >
+      {getName(drPolicy)}
+    </DropdownItem>
+  ));
+
+  const onToggle = (isOpen: boolean) => {
+    setIsOpen(isOpen);
+  };
+
+  const onSelect = (e) => {
+    // Selecting a DRPolicy to intiate failover or relocate
+    const drPolicy: DRPolicyKind = drPolicies?.[e.currentTarget.id];
+    setSelected({
+      policyName: getName(drPolicy),
+      drClusters: drPolicy?.spec?.drClusters,
+    });
```

Comment:
nits: `?.` not required
```suggestion
    const drPolicy: DRPolicyKind = drPolicies[e.currentTarget.id];
    setSelected({
      policyName: getName(drPolicy),
      drClusters: drPolicy.spec?.drClusters,
    });
```